lib/YaST/Firstboot/FirstbootController.pm

```diff
@@ -77,4 +85,17 @@ sub press_next {
     $self->get_generic_page->press_next();
 }
 
+sub return_language_and_keyboard {
+    my ($self) = @_;
+    my %settings;
+    $settings{language} = $self->get_language_and_keyboard_page->return_selected_language();
```

Comment:
just a hint, if you use reference values here, you can just move it as scalars among methods, i.e.: `my $settings`; $settings->{language}`. Or just at the end add `return {%settings}` or `\%settings`;